app/recurring_job.go

```diff
@@ -436,29 +415,27 @@ func (job *Job) doSnapshotCleanup(backupDone bool) (err error) {
 		}
 	}()
 
-	volumeAPI := job.api.Volume
 	volumeName := job.volumeName
-	volume, err := volumeAPI.ById(volumeName)
+	volume, err := job.api.Volume.ById(volumeName)
 	if err != nil {
 		return errors.Wrapf(err, "could not get volume %v", volumeName)
 	}
 
-	collection, err := volumeAPI.ActionSnapshotList(volume)
+	collection, err := job.api.Volume.ActionSnapshotCRList(volume)
 	if err != nil {
 		return err
 	}
 
-	deleteSnapshotNames := job.listSnapshotNamesToCleanup(collection.Data, backupDone)
-	if err := job.deleteSnapshots(deleteSnapshotNames, volume, volumeAPI); err != nil {
-		return err
-	}
-
-	shouldPurgeSnapshots := len(deleteSnapshotNames) > 0 || job.task == longhorn.RecurringJobTypeSnapshotCleanup || job.task == longhorn.RecurringJobTypeSnapshotDelete
-	if shouldPurgeSnapshots {
-		if err := job.purgeSnapshots(volume, volumeAPI); err != nil {
```

Comment:
How do Longhorn recurring jobs handle and trigger purge actions now?

Reply:
It is handled in the snapshot controller. When the snapshot CR is deleted, snapshot controller will try to do purge to remove the snapshot